modules/nft/keeper/collection.go

```diff
@@ -34,85 +27,42 @@ func (k Keeper) SetCollection(ctx sdk.Context, collection types.Collection) erro
 
 // GetCollection returns the collection by the specified denom ID
 func (k Keeper) GetCollection(ctx sdk.Context, denomID string) (types.Collection, error) {
-	denom, found := k.GetDenom(ctx, denomID)
-	if !found {
-		return types.Collection{}, sdkerrors.Wrapf(types.ErrInvalidDenom, "denomID %s not existed ", denomID)
+	denom, err := k.GetDenomInfo(ctx, denomID)
+	if err != nil {
+		return types.Collection{}, err
 	}
 
-	nfts := k.GetNFTs(ctx, denomID)
-	return types.NewCollection(denom, nfts), nil
-}
-
-// GetPaginateCollection returns the collection by the specified denom ID
-func (k Keeper) GetPaginateCollection(ctx sdk.Context, request *types.QueryCollectionRequest, denomID string) (types.Collection, *query.PageResponse, error) {
-	denom, found := k.GetDenom(ctx, denomID)
-	if !found {
-		return types.Collection{}, nil, sdkerrors.Wrapf(types.ErrInvalidDenom, "denomID %s not existed ", denomID)
-	}
-	var nfts []exported.NFT
-	store := ctx.KVStore(k.storeKey)
-	nftStore := prefix.NewStore(store, types.KeyNFT(denomID, ""))
-	pageRes, err := query.Paginate(nftStore, shapePageRequest(request.Pagination), func(key []byte, value []byte) error {
-		var baseNFT types.BaseNFT
-		k.cdc.MustUnmarshal(value, &baseNFT)
-		nfts = append(nfts, baseNFT)
-		return nil
-	})
+	nfts, err := k.GetNFTs(ctx, denomID)
 	if err != nil {
-		return types.Collection{}, nil, status.Errorf(codes.InvalidArgument, "paginate: %v", err)
+		return types.Collection{}, err
 	}
-	return types.NewCollection(denom, nfts), pageRes, nil
+	return types.NewCollection(*denom, nfts), nil
 }
 
```

Comment:
Is `GetCollection` necessary anymore since the REST query has been deprecated?

Reply:
`GetPaginateCollection` was only called by `Collection` in gRPC query, so we don't need it anymore?

Reply:
> Is `GetCollection` necessary anymore since the REST query has been deprecated?

deleted

Reply:
> `GetPaginateCollection` was only called by `Collection` in gRPC query, so we don't need it anymore?

yes